extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/OpenTelemetryEmitter.java

```diff
@@ -0,0 +1,118 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class OpenTelemetryEmitter implements Emitter
+{
+  private static final Logger log = new Logger(OpenTelemetryEmitter.class);
+  private static final DruidContextTextMapGetter DRUID_CONTEXT_MAP_GETTER = new DruidContextTextMapGetter();
+  private final Tracer tracer;
+
+  OpenTelemetryEmitter(Tracer tracer)
+  {
+    this.tracer = tracer;
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting OpenTelemetryEmitter");
+  }
+
+  @Override
+  public void emit(Event e)
+  {
+    if (!(e instanceof ServiceMetricEvent)) {
+      return;
+    }
+    ServiceMetricEvent event = (ServiceMetricEvent) e;
+
+    // We generate spans for the following types of events:
+    // query/time
+    if (!event.getMetric().equals("query/time")) {
+      return;
+    }
+
+    emitQueryTimeEvent(event);
+  }
+
+  private void emitQueryTimeEvent(ServiceMetricEvent event)
+  {
+    Context context = GlobalOpenTelemetry.getPropagators()
+                                         .getTextMapPropagator()
+                                         .extract(Context.current(),
+                                                  getContextAsString(event),
+                                                  DRUID_CONTEXT_MAP_GETTER
+                                         );
+
+    try (Scope scope = context.makeCurrent()) {
+      DateTime endTime = event.getCreatedTime();
+      DateTime startTime = endTime.minusMillis(event.getValue().intValue());
+
+      Span span = tracer.spanBuilder(event.getService()).setStartTimestamp(
+          startTime.getMillis(),
+          TimeUnit.MILLISECONDS
+      ).startSpan();
+      span.end(endTime.getMillis(), TimeUnit.MILLISECONDS);
```

Comment:
why not chain the `.end()` call with the rest instead of creating a `span` variable?

Reply:
Oh, thanks!
I left it because here we can add attributes to a span
Which attributes do we want to add? @xvrl @Zohimi @sirianni 
I think, we should add queryId, but don't know about others

Reply:
Now added `.end` call in the chain
We can change later